webapp/channels/src/actions/notification_actions.jsx

```diff
@@ -265,12 +273,22 @@ export function sendDesktopNotification(post, msgProps) {
         const channelId = channel ? channel.id : null;
 
         let notify = false;
+        let notifyResult = {result: 'not_sent', reason: 'unknown'};
         if (isCrtReply) {
             notify = !isThreadOpen(state, post.root_id);
+            if (!notify) {
+                notifyResult = {result: 'not_sent', reason: 'thread_is_open', data: post.root_id};
+            }
         } else {
             notify = activeChannel && activeChannel.id !== channelId;
+            if (!notify) {
+                notifyResult = {result: 'not_sent', reason: 'channel_is_open', data: activeChannel?.id};
+            }
         }
         notify = notify || !state.views.browser.focused;
+        if (!notify) {
```

Comment:
I'm pretty sure that this always overwrites the`notifyResult` from above, but I'm not entirely sure given how weird this chunk of the code is

Reply:
GIven the above statement, if `notify` is `true` and `state.views.browser.focused` is `true`, that's the only case where this will be overwritten, which is the exact case described in the `notifyResult`. Definitely worth tracking to help point out notifications that people may be "missing" when really they just had the app open still.

Reply:
I don't think that's right. If `notify` was true before the above line, then it would still be true after `notify || !state.views.browser.focused`, so we'd never enter this if block.

The above line would only result in `notify` being false if it was already false and `state.views.browser.focused` is true, and if it was already false, then `notifyResult` would have already been set on line 280 or 285 in the if-else above, so it's getting overwritten here.

I'm having a hell of a time even parsing what the original code was meant to do, so can we change these lines to the following?
```js
if (!notify && !state.views.browser.focused) {
    notify = true;
    notifyResult = {...};
}
```
I think that's logically equivalent to what the old code is doing and what you're trying to accomplish

Reply:
Actually, no, that's not even right. The previous if-else block is making it so that you don't receive a notification if you have the channel/thread open that contains the new post and then the `notify || !state.views.browser.focused` line is saying "if you have that channel/thread open but the window is in the background, still send a notification". Holy crap, this logic is confusing.

Since the only change there is that we're sending a notification when we wouldn't otherwise have, there's no reason to set a `notifyResult`. If this makes sense to you, you should remove this if block since the `window_is_focused` case is already handled by `channel_is_open` and `thread_is_open`